test/functional/lib/ipfs-companion.test.js

```diff
@@ -1,33 +1,37 @@
-const { describe, it, before, beforeEach, after } = require('mocha')
-const { expect } = require('chai')
-const browser = require('sinon-chrome')
-const { URL } = require('url')
-const { optionDefaults } = require('../../../add-on/src/lib/options')
-const { AbortController } = require('abort-controller')
+import { describe, it, before, after } from 'mocha'
+import { expect } from 'chai'
+import browser from 'sinon-chrome'
+import AbortController from 'abort-controller'
+import { URL } from 'url'
+import { optionDefaults } from '../../../add-on/src/lib/options.js'
+browser.runtime.id = 'testid'
+global.browser = browser
+global.AbortController = AbortController
+global.chrome = browser
+global.navigator = {
+  clipboard: {
+    writeText: () => {}
+  }
+}
+// We need to do this because global is not mapped otherwise, we need to stub browser and chrome runtime
+// so that the webextension-polyfill does not complain about the test runner not being a browser instance.
+const init = async () => (await import('../../../add-on/src/lib/ipfs-companion.js')).default()
```

Comment:
this was really the problem I had to solve because how node's module resolution works. stubbing `global.browser` was not possible when `import` for a module is called.

However doing it this way allows us to pass in stubbed references.